src/core/counter.h

```diff
@@ -0,0 +1 @@
+struct counter { uint64_t c; };
```

Comment:
don't think so... in `link.lua` the `require('core.link_h')` is after the `require('core.counter')`, so it should work without pulling this specifically.
In fact, unless a struct or const is used in a C function, I see no value in using `.h` files.  Maybe somebody dislike the `ffi.cdef[[...]]` embedded definitions?
The only C code I've written since we pulled ljsyscall was for the pthreads experiment, but that was very short-lived.  The current `fork()`based doesn't need a single `.c` file.